arch/risc-v/Kconfig

```diff
@@ -337,7 +337,7 @@ config ARCH_RV32
 config ARCH_RV64
```

Comment:
should we add a new ARCH type to avoid the change couple with a specific toolchain(RISCV_TOOLCHAIN_GNU_RV64ILP32)?

Reply:
It shares the RV64 instruction set so still belongs to ARCH_RV64, treating it as a new toolchain looks simpler. We just need stop assuming that `uintptr_t` is of register width.

Reply:
but how to handle if clang support ILP32?

Reply:
I saw current `Toolchain.defs` or `Toolchain.cmake` already using `CONFIG_RISCV_TOOLCHAIN_CLANG`, so looks like Clang is already treated as a toolchain, not as a new architecture?

Reply:
The check of toolchain specific option should be limited inside Toolchain.defs/Toolchain.cmake, it's bad to check them outside these two files.

Reply:
Thanks, I will see if a `ARCH_RV6ILP32` helps to make C sources cleaner.

Reply:
Thanks, I will see if a `ARCH_RV6ILP32` helps to make C sources cleaner.

Reply:
An `ARCH_RV64ILP32` was added for use in C sources.